js/digifinex.js

```diff
@@ -1031,23 +1137,79 @@ module.exports = class digifinex extends Exchange {
         //         "is_maker": true
         //     }
         //
-        const id = this.safeString (trade, 'id');
+        // swap: fetchMyTrades
+        //
+        //     {
+        //         "trade_id": "1590136768424841218",
+        //         "instrument_id": "BTCUSDTPERP",
+        //         "order_id": "1590136768156405760",
+        //         "type": 1,
+        //         "order_type": 8,
+        //         "price": "18514.5",
+        //         "size": "1",
+        //         "fee": "0.00925725",
+        //         "close_profit": "0",
+        //         "leverage": "20",
+        //         "trade_type": 0,
+        //         "match_role": 1,
+        //         "trade_time": 1667953123562
+        //     }
+        //
+        const id = this.safeString2 (trade, 'id', 'trade_id');
         const orderId = this.safeString (trade, 'order_id');
-        const timestamp = this.safeTimestamp2 (trade, 'date', 'timestamp');
-        let side = this.safeString2 (trade, 'type', 'side');
-        const parts = side.split ('_');
-        side = this.safeString (parts, 0);
-        const type = this.safeString (parts, 1);
         const priceString = this.safeString (trade, 'price');
-        const amountString = this.safeString (trade, 'amount');
-        const marketId = this.safeString (trade, 'symbol');
-        const symbol = this.safeSymbol (marketId, market, '_');
-        const takerOrMaker = this.safeValue (trade, 'is_maker');
-        const feeCostString = this.safeString (trade, 'fee');
+        const amountString = this.safeStringN (trade, [ 'amount', 'volume', 'size' ]);
+        const marketId = this.safeStringUpper2 (trade, 'symbol', 'instrument_id');
+        const symbol = this.safeSymbol (marketId, market);
+        if (market === undefined) {
+            market = this.safeMarket (marketId);
+        }
+        let timestamp = this.safeTimestamp2 (trade, 'date', 'timestamp');
+        let side = this.safeString2 (trade, 'type', 'side');
+        let type = undefined;
+        let takerOrMaker = undefined;
+        if (market['type'] === 'swap') {
+            timestamp = this.safeInteger (trade, 'trade_time');
+            const orderType = this.safeString (trade, 'order_type');
+            const tradeRole = this.safeString (trade, 'match_role');
+            const direction = this.safeString (trade, 'direction');
+            if (orderType !== undefined) {
+                type = (orderType === '0') ? 'limit' : undefined;
+            }
+            if (tradeRole === '1') {
+                takerOrMaker = 'taker';
+            } else if (tradeRole === '2') {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = undefined;
+            }
+            if ((side === '1') || (direction === '1')) {
+                side = 'open long';
```

Comment:
@Dan-krm, these are more informative for sure, but shouldn't we be using the unified side here? (buy/sell)